nav2_straightline_planner/include/nav2_straightline_planner/rrt_planner.hpp

```diff
@@ -0,0 +1,367 @@
+#pragma once
+
+#include <string>
+#include <vector>
+#include <cstdlib>
+#include <ctime>
+#include <random>
+#include <cmath>
+#include <iostream> // for std::cout
+
+#include <nav_msgs/msg/path.hpp>
+#include <nav_msgs/msg/occupancy_grid.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <std_msgs/msg/string.hpp>
+#include "nav2_costmap_2d/costmap_2d_ros.hpp"
+
+namespace nav2_straightline_planner
+{   
+    // from isc_nav/include/utility/point.hpp
+    struct Point2D
+    {
+        double x, y;
+        Point2D( const double& ix, const double& iy )
+            : x( ix )
+            , y( iy )
+        {
+        }
+        Point2D() {}
+    };
+
+    // from isc_nav/include/utility/point.hpp
+    inline bool operator==( const Point2D& lhs, const Point2D& rhs )
+    {
+        return ( lhs.x == rhs.x && lhs.y == rhs.y );
+    }
+    
+    class RRT 
```

Comment:
header file should only include class and method definitions. cpp file will include the actually implementation of the methods